onnxruntime/core/providers/cuda/cuda_pch.h

```diff
@@ -14,6 +14,21 @@
 #include <curand.h>
 #include <cudnn.h>
 
+// support of cublasLt starts 10.1
+#if CUDA_VERSION >= 10010
+#include <cublasLt.h>
+#else
+typedef void* cublasLtHandle_t;
+
+inline cublasStatus_t cublasLtCreate(cublasLtHandle_t* /*handle*/) {
+  return CUBLAS_STATUS_SUCCESS;
```

Comment:
It might be better to return failure in this function, and avoid calling this function like the following:

#if CUDA_VERSION >= 10010
    cublasLtCreate(handle)
#endif

Reply:
Those function/type subs are to avoid adding "if CUDA_VERSION >= 10010" check in other places, like in cuda_common.h, cuda_execution_provider.h. It's better to return a failure status.

---
In reply to: [404464744](https://github.com/microsoft/onnxruntime/pull/3413#discussion_r404464744) [](ancestors = 404464744)